modules/mod_mailinglist/models/m_mailinglist.erl

```diff
@@ -202,7 +202,7 @@ recipient_delete1(RecipientProps, Quiet, Context) ->
 recipient_confirm(ConfirmKey, Context) ->
 	case z_db:q_row("select id, is_enabled, mailinglist_id from mailinglist_recipient where confirm_key = $1", [ConfirmKey], Context) of
 		{RecipientId, _IsEnabled, ListId} ->
-			NewConfirmKey = z_ids:id(20),
+			NewConfirmKey = binary_to_list(z_ids:id(20)),
 			z_db:q("update mailinglist_recipient set confirm_key = $2, is_enabled = true where confirm_key = $1", [ConfirmKey, NewConfirmKey], Context),
```

Comment:
I was not sure about the effects. 

I don't know what conversions happen when you store and retrieve a value in the database. If a string is automatically converted to a binary when stored, then obviously there is no reason to convert it to string first, and the call to binary_to_list() can be removed.

In general I tried to "err on the safe side", and I added conversions to the original string type where it was not clear to me that it would be safe to omit them.